changelogs/fragments/fix-snmp-server-idempotency.yml

```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - Fixed idempotency issue in SNMP configuration management. Replaced in-place modifications of `i_have` with key tracking using a set. This change preserves the original `i_have` during iteration, ensuring consistent and accurate updates by only negating the remaining keys and preventing unintended overwrites.
```

Comment:
This is too much information, you can leave out the technical elements and put in whats error/bug it is solving, this will ultimately go in the release changelog, hence we have to be more careful about it

Reply:
Updated in the new PR